src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+        path, path_parameters.t_start, path_parameters.t_end, arc_length_param);
+
+    Trajectory_t const_interp_trajectory;
+    TrajectoryElement_t const_interp_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    const_interp_trajectory.trajectory_elements = const_interp_elements;
+
+    // Calculate the constant-interpolation period equivalent of the trajectory
+    app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation(
+        &const_interp_trajectory, &trajectory, 0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.x,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.x,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.y,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.y,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].time,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .time,
+                0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.x,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.y,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].time,
+                const_interp_trajectory.trajectory_elements[0].time, 0.001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_assert_cannot_reach_final_velocity)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 1;
+    path_parameters.num_segments               = 5999;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 2000;
+    path_parameters.initial_speed              = 0;
+    path_parameters.final_speed                = 3000;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+
+    enum TrajectoryPlannerGenerationStatus status =
+        app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                     &trajectory);
+
+    EXPECT_EQ(finalVelocityTooHigh, status);
+}
+
+TEST_F(TrajectoryPlannerTest, test_assert_initial_velocity_too_high)
```

Comment:
A quick explanation here would be helpful like
"Set the initial speed impossibly large so that the robot should not be able to follow a very curved path"

Reply:
Added pre-comment